src/os/unix/wayland.rs

```diff
@@ -0,0 +1,1182 @@
+        self.toplevel.set_min_size(size.0, size.1);
+    }
+
+    //Sets a specific cursor style
+    fn update_cursor(&mut self, cursor: &str) {
+        let csr = self.cursor.get_cursor(cursor).unwrap();
+        let img = csr.frame_buffer(0).unwrap();
+        self.cursor_surface.attach(Some(&*img), 0, 0);
+        self.cursor_surface.damage(0, 0, 32, 32);
+        self.cursor_surface.commit();
+    }
+
+    //resizes when buffer is bigger or less
+    fn update_framebuffer(&mut self, buffer: &[u32], size: (i32, i32)) {
+        use std::io::{Seek, SeekFrom};
+
+        let (mut fd, buf) = self.buf_pool.get_buffer(size);
+
+        fd.seek(SeekFrom::Start(0)).unwrap();
+        let slice = unsafe {
+            std::slice::from_raw_parts(
+                buffer[..].as_ptr() as *const u8,
+                buffer.len() * std::mem::size_of::<u32>(),
+            )
+        };
+        fd.write_all(&slice[..]).unwrap();
+        fd.flush().unwrap();
+
+        //Acknowledge the last configure event
+        if let Some(serial) = (*self.xdg_config.borrow_mut()).take() {
+            self.xdg_surface.ack_configure(serial);
+        }
+
+        self.surface.attach(Some(buf), 0, 0);
+        self.surface
+            .damage(0, 0, i32::max_value(), i32::max_value());
+        self.surface.commit();
+    }
+
+    //Keyboard, Pointer, Touch
+    fn _check_capabilities(
+        seat: &Main<WlSeat>,
+        event_queue: &mut EventQueue,
+    ) -> (bool, bool, bool) {
+        let keyboard_fl = Rc::new(RefCell::new(false));
+        let pointer_fl = Rc::new(RefCell::new(false));
+        let touch_fl = Rc::new(RefCell::new(false));
+
+        {
+            let keyboard_fl = keyboard_fl.clone();
+            let pointer_fl = pointer_fl.clone();
+            let touch_fl = touch_fl.clone();
+
+            //check pointer and mouse capability
+            seat.quick_assign(move |_, event, _| {
+                use wayland_client::protocol::wl_seat::{Capability, Event};
+
+                if let Event::Capabilities { capabilities } = event {
+                    if !*pointer_fl.borrow() && capabilities.contains(Capability::Pointer) {
+                        *pointer_fl.borrow_mut() = true;
+                    }
+                    if !*keyboard_fl.borrow() && capabilities.contains(Capability::Keyboard) {
+                        *keyboard_fl.borrow_mut() = true;
+                    }
+                    if !*touch_fl.borrow() && capabilities.contains(Capability::Touch) {
+                        *touch_fl.borrow_mut() = true;
+                    }
+                }
+            });
+        }
+
+        event_queue
+            .sync_roundtrip(&mut (), |_, _, _| {})
+            .map_err(|e| Error::WindowCreate(format!("Roundtrip failed: {:?}", e)))
+            .unwrap();
+
+        let ret = (
+            *keyboard_fl.borrow(),
+            *pointer_fl.borrow(),
+            *touch_fl.borrow(),
+        );
+
+        ret
```

Comment:
```suggestion
        (*keyboard_fl.borrow(), *pointer_fl.borrow(), *touch_fl.borrow())
```

(or however `rustfmt` decides to format it)